<a href="https://colab.research.google.com/github/vannessa01/Python_Journey/blob/main/Introduction_to_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introduction to Python

- Nama: Vannessa Yoanita Oktorio Naibaho

In [1]:
import numpy as np

# please import as much as you need

# Normal

## 1. Operasi Variabel

> Misal kita memiliki toko sepatu. Modal membeli 1 kodi sepatu adalah Rp 1.000.000. Setiap sepatu dijual seharga Rp 80.000. Buatlah variabel modal, kuantitas, dan harga_jual sesuai informasi di atas. Hitung profit yang didapatkan jika semua sepatu berhasil terjual, dan masukkan ke variabel profit.

Saran langkah-langkah pengerjaan
1. Convert terlebih dahulu kodi dalam jumlah barang (1 kodi = xyz buah sepatu)
2. Gunakan operasi matematika sederhana untuk menghitung profit menggunakan tiga buah variabel yang sudah ditentukan

In [2]:
# soal no 1
modal = 1000000
kuantitas = 20 #1 kodi = 20 buah
harga_jual = 80000

In [3]:
#hitung profit
profit = harga_jual*kuantitas-modal
print(profit)

600000


Jadi, dari profit di atas, dapat disimpulkan bahwa keuntungan yang diperoleh dari hasil penjualan sepatu sebesar 600000 rupiah.
hasil ini diperoleh dengan mengalikan harga jual dengan kuantitas kemudian dikurangi dengan modal

## 2. Package Python dan Dictionary

> Kita akan coba mengakses data scientific article dari PLOS corpus
    Menggunakan API request dengan kode berikut : 
```
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
```
**perhatikan tanda kutipnya tidak salah ketika meng-copy ke dalam python

Pertanyaannya : 
Convert response.content dari request tersebut dari JSON Format ke dalam Dictionary, dan print-kan key di dalamnya apa saja

Saran langkah-langkah pengerjaan
1. Ikuti code yang tertera, kemudian ekstrak value dari variable response menggunakan dot (.) content menjadi sebuah variabel baru (dictionary)
2. Gunakan library json, convert hasil content menggunakan json.loads
3. Print keys dari dictionary tersebut


In [ ]:
# soal no 2
import requests

response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
print(response)

<Response [200]>


In [ ]:
dataset = response.content

In [ ]:
import json

In [ ]:
json.loads(dataset)

In [ ]:
keys_dataset = dict.keys(json.loads(dataset))

In [ ]:
print(keys_dataset)

dict_keys(['response'])


Jadi, dari url link di atas, kami menyimpulkan bahwa memiliki keys 'response'

# Intermediate

## 3. Operasi dalam List

> Buatlah list dengan 10 nama kota di indonesia, kemudian:
1. Urutkan nama kota di dalam list berdasarkan abjad (a)
2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)

Saran langkah-langkah pengerjaan
1. Tidak ada ketentuan nama kota (misal berdasarkan provinsi X atau pulau Y), dibebaskan untuk memilih. Buat nama-nama kota tersebut dalam sebuah list.
2. Gunakan fungsi sorted(X) untuk mengurutkan secara kecil-ke-besar (sesuai abjad), dimana X adalah contoh variabel list pada langkah (a)
3. Gunakan hasil poin (a), urutkan nama kota di dalam list berdasarkan abjad, untuk mengerjakan poin (b) dan (c). 

In [ ]:
# your code goes here!
kota = ["Bogor", "Sukabumi", "Cirebon", "Semarang", "Tegal", "Malang", "Yogyakarta", "Kediri", "Probolinggo", "Jakarta"]
kota

['Bogor',
 'Sukabumi',
 'Cirebon',
 'Semarang',
 'Tegal',
 'Malang',
 'Yogyakarta',
 'Kediri',
 'Probolinggo',
 'Jakarta']

In [ ]:
# soal no 3
kota_urut = sorted(kota)
kota_urut

['Bogor',
 'Cirebon',
 'Jakarta',
 'Kediri',
 'Malang',
 'Probolinggo',
 'Semarang',
 'Sukabumi',
 'Tegal',
 'Yogyakarta']

In [ ]:
#soal b ambil 4 kota pertama dari list yang sudah diurutkan
kota_urut[0:4]

['Bogor', 'Cirebon', 'Jakarta', 'Kediri']

In [ ]:
#soal c ambil 3 kota terakhir dari list yang sudah diurutkan
kota_urut[7:11] # atau kota_urut[7:10]

['Sukabumi', 'Tegal', 'Yogyakarta']

Jadi, dari operasi list di atas, dapat disimpulkan bahwa
a. urutan kota sesuai abjad adalah Bogor, Cirebon, Jakarta, Kediri, Malang, Probolinggo,Semarang,Sukabumi, Tegal, Yogyakarta
b. 4 kota pertama dari list yang sudah diurutkan adalah Bogor, Cirebon, Jakarta, Kediri
c. 3 kota terakhir dari list yang sudah diurutkan adalah Sukabumi, Tegal, Yogyakarta

## 4. Function dalam List

> Team sales memiliki list total transaksi dari 10 merchant pada bulan Juni 2020, yaitu: 
```
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
```
1. Trx minimum
2. Rata-rata 
3. Trx maksimal

Saran langkah-langkah pengerjaan
1. Copy variabel total_transaksi yang ada di soal ke notebook kita (atau ketik ulang boleh)
2. Kemudian, hitung menggunakan function operasi dalam list yang telah dipelajari di kelas pada topik list. Print hasilnya setiap poin (a), (b), dan (c)

In [ ]:
# soal no 4
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
#point a
print(min(total_transaksi))
#point b
print(sum(total_transaksi)/len(total_transaksi))
#point c
print(max(total_transaksi))

10
25.6
45


Jadi, dari hasil operasi list di atas, dapat disimpulkan bahwa nilai minimal: 10, max: 45, rata-rata: 25.6

# Advanced

## 5. NumPy Array

> Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1), Kemudian lakukan perintah berikut : 
1. Kali-kan array tersebut dengan 10
2. Bagi-kan array (hasil dari bagian a) dengan 3
3. Kali-kan array (hasil dari bagian b) dengan array baru :
`[-45, 43, 3, 56, 5]`
4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c) 

Saran langkah-langkah pengerjaan
1. HATI-HATI dalam pengerjaan, karena apabila poin sebelumnya salah, maka selanjutnya akan kebawa salah (misal (b) salah, maka poin (c) sampai (d) akan salah)
2. Untuk array 5 bilangan ganjil pertama boleh manual boleh menggunakan fungsi, dimulai dari 1
3. Ikuti perintah soal, gunakan numpy array, jangan list. Sesuaikan dengan permintaan soal pada setiap butir.



In [ ]:
# soal no 5
import numpy as np

In [ ]:
bilangan_ganjil = np.array([1, 3, 5, 7, 9])
bilangan_ganjil

array([1, 3, 5, 7, 9])

In [ ]:
# a.Kali-kan array tersebut dengan 10
x = bilangan_ganjil * 10
x

array([10, 30, 50, 70, 90])

In [ ]:
# b.Bagi-kan array (hasil dari bagian a) dengan 3
y = x / 3
y

array([ 3.33333333, 10.        , 16.66666667, 23.33333333, 30.        ])

In [ ]:
# c. Kali-kan array (hasil dari bagian b) dengan array baru : [-45, 43, 3, 56, 5]
v = np.array([-45, 43, 3, 56, 5])
v
z  = y * v
z

array([-150.        ,  430.        ,   50.        , 1306.66666667,
        150.        ])

In [ ]:
# d. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c)
index_max = np.argmax(z)
index_max

3

In [ ]:
index_min = np.argmin(z)
index_min

0

Jadi, dari hasil operasi numpy di atas, dapat disimpulkan bahwa a. hasil perkalian dengan array tersebut dengan 10 adalah 10, 30, 50, 70, 90. b. hasil pembagian array (hasil dari bagian a) dengan 3 adalah  3.33333333, 10, 16.66666667, 23.33333333, 30. c. hasil perkalian array (hasil dari bagian b) dengan array baru adalah -150, 430, 50, 1306.66666667, 150. d. index max: 3, index min: 0.

# Terimakasih, salam dari Bogor